File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/infrastruktur/api/OrganisasjonerController.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.infrastruktur.api;

import java.util.List;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.infrastruktur.altinn.AltinnOrganisasjon;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.domenemodeller.InnloggetBruker;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.TilgangskontrollService;
import no.nav.security.token.support.core.api.Protected;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@Protected
@RestController
public class OrganisasjonerController {

  private final TilgangskontrollService tilgangskontrollService;

  public OrganisasjonerController(TilgangskontrollService tilgangskontrollService) {
    this.tilgangskontrollService = tilgangskontrollService;
  }

  @GetMapping("/organisasjoner/statistikk")
  public List<AltinnOrganisasjon> hentOrganisasjonerMedStatistikktilgang() {
    InnloggetBruker innloggetBruker = tilgangskontrollService.hentBrukerKunIaRettigheter();
    return innloggetBruker.getBrukerensOrganisasjoner();
  }

  @GetMapping("/organisasjoner")
  public List<AltinnOrganisasjon> hentOrganisasjonerMedAlleTilganger() {
    InnloggetBruker innloggetBruker =
        tilgangskontrollService.hentInnloggetBrukerForAlleRettigheter();
    return innloggetBruker.getBrukerensOrganisasjoner();
  }
}

File: src/main/kotlin/no/nav/arbeidsgiver/sykefravarsstatistikk/api/infrastruktur/api/OrganisasjonerController.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.infrastruktur.api

import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.tilgangsstyring.TilgangskontrollService
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.infrastruktur.altinn.AltinnOrganisasjon
import no.nav.security.token.support.core.api.Protected
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@Protected
@RestController
class OrganisasjonerController(private val tilgangskontrollService: TilgangskontrollService) {
    @GetMapping("/organisasjoner/statistikk")
    fun hentOrganisasjonerMedStatistikktilgang(): List<AltinnOrganisasjon> {
        val (_, brukerensOrganisasjoner) = tilgangskontrollService.hentBrukerKunIaRettigheter()
        return brukerensOrganisasjoner
    }

    @GetMapping("/organisasjoner")
    fun hentOrganisasjonerMedAlleTilganger(): List<AltinnOrganisasjon> {
        val (_, brukerensOrganisasjoner) = tilgangskontrollService.hentInnloggetBrukerForAlleRettigheter()
        return brukerensOrganisasjoner
    }
}
